.changeset/spicy-pumas-suffer.md

```diff
@@ -0,0 +1,65 @@
+---
+'@makeswift/controls': minor
+---
+
+This change converts our control definitions to active classes, each with
+encapsulated implementations for control serialization/deserialization, data and
+value parsing, copying, and other core operations for controls.
+
+In addition, base definitions of all controls, as well as several shared
+types/schemas, have now been defined in this package.
+
+## BREAKING:
+
+1. Several types and utilities associated with controls have been removed as part of
+   this change. As an example, let's look over the types/functions associated with
+   the `Color` control (keep in mind that this applies to all controls, not just
+   `Color`):
+
+   - `ColorControlType`: a string literal identifying the `Color` control
+
+   - `ColorControlDataTypeValueV1` - a string literal identifying V1 Color control
+     data.
+
+   - `ColorControlDataTypeKey`: a string literal defining the key in which the V1
+     data signature for the Color control is stored in.
+
+   - `ColorControlDataV0` / `ColorControlDataV1` - types describing the data format
+     of the color control, depending on the version of the definition.
+
+   - `ColorControlData` - describes the data format for a `Color` control. In this
+     case, it is a union of type definitions for `ColorControlDataV0` and
+     `ColorControlDataV1`
+
+   - `ColorControlDefinition` - describes the type of the `Color` control
+     definition, namely its type and config options
+
+   - `copyColorData` - Creates a copy of a given `ColorControlData`, given a copy
+     context.
+
+   All of these have been removed. Most of them have been replaced by
+   functionality available on the `ColorDefinition` active class. When we
+   register our components, we are creating an instance of a `ColorDefinition`
+   class. Let's see how we can use this class to achieve the functionality of
+   the removed exports above:
+
+   ```typescript
+   import { DataType } from './associated-types'
+   import { Color, ColorDefinition } from './color'
```

Comment:
Ah, I defined these from the perspective of an internal user, not an external one (since this is meant for internal use only). That's not appropriate for a changelog, so I'll revert to what it was.